packages/jest-snapshot/src/__tests__/dedentLines.test.ts

```diff
@@ -0,0 +1,268 @@
+  });
+
+  test('one line empty object', () => {
+    const val = {};
+    const indented = formatLines2(val);
+    const dedented = formatLines0(val);
+
+    expect(dedentLines(indented)).toEqual(dedented);
+  });
+
+  test('one line self-closing element', () => {
+    const val = {
+      $$typeof,
+      children: null,
+      type: 'br',
+    };
+    const indented = formatLines2(val);
+    const dedented = formatLines0(val);
+
+    expect(dedentLines(indented)).toEqual(dedented);
+  });
+
+  test('object value empty string', () => {
+    const val = {
+      key: '',
+    };
+    const indented = formatLines2(val);
+    const dedented = formatLines0(val);
+
+    expect(dedentLines(indented)).toEqual(dedented);
+  });
+
+  test('object value string includes double-quote marks', () => {
+    const val = {
+      key: '"Always bet on JavaScript",',
+    };
+    const indented = formatLines2(val);
+    const dedented = formatLines0(val);
+
+    expect(dedentLines(indented)).toEqual(dedented);
+  });
+
+  test('markup with props and text', () => {
+    const val = {
+      $$typeof,
+      children: [
+        {
+          $$typeof,
+          props: {
+            alt: 'Jest logo',
+            src: 'jest.svg',
+          },
+          type: 'img',
+        },
+        {
+          $$typeof,
+          children: ['Delightful JavaScript testing'],
+          type: 'h2',
+        },
+      ],
+      type: 'header',
+    };
+    const indented = formatLines2(val);
+    const dedented = formatLines0(val);
+
+    expect(dedentLines(indented)).toEqual(dedented);
+  });
+
+  test('markup with components as props', () => {
+    // https://daveceddia.com/pluggable-slots-in-react-components/
+    const val = {
+      $$typeof,
+      children: null,
+      props: {
+        content: {
+          $$typeof,
+          children: ['main content here'],
+          type: 'Content',
+        },
+        sidebar: {
+          $$typeof,
+          children: null,
+          props: {
+            user: '0123456789abcdef',
+          },
+          type: 'UserStats',
+        },
+      },
+      type: 'Body',
+    };
+    const indented = formatLines2(val);
+    const dedented = formatLines0(val);
+
+    expect(dedentLines(indented)).toEqual(dedented);
+  });
+});
+
+describe('dedentLines null', () => {
+  test('object key multi-line', () => {
+    const val = {
+      'multi\nline\nstring': false,
+    };
+    const indented = formatLines2(val);
+
+    expect(dedentLines(indented)).toEqual(null);
+  });
+
+  test('object value multi-line', () => {
+    const val = {
+      key: 'multi\nline\nstring',
+    };
+    const indented = formatLines2(val);
+
+    expect(dedentLines(indented)).toEqual(null);
+  });
+
+  test('object key and value multi-line', () => {
+    const val = {
+      'multi\nline\nstring': '\nleading newline',
+    };
+    const indented = formatLines2(val);
+
+    expect(dedentLines(indented)).toEqual(null);
```

Comment:
There's quite a bunch of repetition in these tests. It's quite readable and I don't say it's bad. However, have you considered using `test.each` where it makes sense (e.g. for smaller inputs)?

Reply:
Oh yeah, super suggestion!